src/R/Components/Impl/InteractiveWorkflow/Implementation/RInteractiveEvaluator.cs

```diff
@@ -78,10 +78,10 @@ public sealed class RInteractiveEvaluator : IInteractiveEvaluator {
             } catch (RHostBinaryMissingException) {
                 await _coreShell.ShowErrorMessageAsync(Resources.Error_Microsoft_R_Host_Missing);
                 return ExecutionResult.Failure;
-            } catch (RHostDisconnectedException) {
-                // We don't 
+            } catch (RHostDisconnectedException rhdex) {
                 return ExecutionResult.Success;
-            } catch (Exception) {
+            } catch (Exception ex) {
+                CurrentWindow.WriteError(Environment.NewLine + ex.Message);
```

Comment:
There is a dilemma with what should be returned - `ExecutionResult.Success` or `ExecutionResult.Failure`. `ExecutionResult.Failure` freezes Interactive Window, forcing user to click Reset. But if connection is lost, Reset won't help. On the other hand, if connection is restored, there is no reason to freeze Interactive Window connection anymore.